spring-kafka-test/src/main/java/org/springframework/kafka/test/EmbeddedKafkaBroker.java

```diff
@@ -417,22 +421,100 @@ private void createTopics(AdminClient admin, List<NewTopic> newTopics) {
 		}
 	}
 
+	/**
+	 * Add topics to the existing broker(s) using the configured number of partitions.
+	 * The broker(s) must be running.
+	 * @param topicsToAdd the topics.
+	 * @return the results; null values indicate success.
+	 * @since 2.5.4
+	 */
+	public Map<String, Exception> addTopicsWithResults(String... topicsToAdd) {
+		Assert.notNull(this.zookeeper, "Broker must be started before this method can be called");
+		HashSet<String> set = new HashSet<>(Arrays.asList(topicsToAdd));
+		this.topics.addAll(set);
+		return createKafkaTopicsWithResults(set);
+	}
+
+	/**
+	 * Add topics to the existing broker(s) and returning a map of results.
+	 * The broker(s) must be running.
+	 * @param topicsToAdd the topics.
+	 * @return the results; null values indicate success.
+	 * @since 2.5.4
+	 */
+	public Map<String, Exception> addTopicsWithResults(NewTopic... topicsToAdd) {
+		Assert.notNull(this.zookeeper, "Broker must be started before this method can be called");
+		for (NewTopic topic : topicsToAdd) {
+			Assert.isTrue(this.topics.add(topic.name()), () -> "topic already exists: " + topic);
+			Assert.isTrue(topic.replicationFactor() <= this.count
+							&& (topic.replicasAssignments() == null
+							|| topic.replicasAssignments().size() <= this.count),
+					() -> "Embedded kafka does not support the requested replication factor: " + topic);
+		}
+
+		return doWithAdminFunction(admin -> createTopicsWithResults(admin, Arrays.asList(topicsToAdd)));
```

Comment:
Why there is no wrapping to `Set` over here?
And with that it looks like we could call the same `createKafkaTopicsWithResults(set)` without this extra lambda...

Reply:
? These are `NewTopic` objects not Strings; `createKafkaTopicsWithResults(set)` takes a set of String (which it converts to a list of `NewTopic`.

I didn't think it was worth trying to weed out duplicates when they provide `NewTopic` objects - because it won't really catch them unless all the propertied are the same.

Reply:
Ah, OK! Missed all those points. 😄 
Merging...